File: 0066: plusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        
        int carry = 1;
        for(int i = 0; i < digits.size(); i++){

            int index = digits.size()-1-i;
            digits[index] += carry;
            carry = 0; //reset 
            if(digits[index] == 10){

                digits[index] = 0;
                carry = 1;
            }
            
            if(index == 0 && carry == 1){

                digits.insert(digits.begin(), 1);
                break;
            }
          
        }

        return digits;
    }
};

File: 0066: plusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        j = len(digits)-1
        carry = 1
        while j >= 0:
            digits[j] += carry
            carry = 0
            if digits[j] >= 10 and j == 0:
                digits[j] = 1
                digits.append(0)
            elif digits[j] >= 10:
                digits[j] = 0
                carry = 1
            j-=1
        return digits